### PR TITLE
Fix master PR debian packaging script

### PR DESCRIPTION
**Pull request by goodlyrottenapple**
A follow up to #1865 fixing the debian packaging script used in the dockerhub upload CI job. To ensure this does not fail again on master merge, the CI job (sans actual upload) was copied to the test PR workflow.

### CODE REVIEW

**Review by Baltoli — APPROVED** [review]
The fix looks good, but remember to remove the dry-run step.

File: package/debian/rules [hunk 1]
```diff
@@ -17,7 +17,7 @@
 	dh $@
 
 override_dh_auto_build:
-	make RELEASE=true plugin-deps kevm-pyk -j3
+	make RELEASE=true CXX=clang++-14 plugin-deps kevm-pyk -j3
```
Comment (Baltoli) [inline package/debian/rules:20]:
```suggestion
	make RELEASE=true CC=clang-14 CXX=clang++-14 plugin-deps kevm-pyk -j3
```

Builds fine as is but for the sake of consistency we can add this

**Review by Baltoli — APPROVED** [review]

**Comment by goodlyrottenapple** [comment]
> LGTM - I presume the intention is to keep the dry-run step in the workflow to prevent breakages in future?

yes, that would be ideal, unless the build times get too long as a result?

**Comment by Baltoli** [comment]
I don't think it'll be the critical component in the parallel build here so probably worth keeping in.

File: .github/workflows/test-pr.yml [hunk 2]
```diff
@@ -204,3 +204,61 @@ jobs:
...
+          k_version=$(cat deps/k_release)
+          container_name="kevm-ci-test-${GITHUB_SHA}"
+          docker cp ./tests/foundry ${container_name}:/home/user/foundry
+          docker exec -u user ${container_name} /bin/bash -c "sudo chown user:user -R /home/user/foundry"
+          docker exec -u user ${container_name} /bin/bash -c "forge build --root foundry"
+          docker exec -u user ${container_name} /bin/bash -c "kevm foundry-kompile --foundry-project-root foundry --verbose"
+          docker exec -u user ${container_name} /bin/bash -c "kevm foundry-prove --foundry-project-root foundry --verbose --test AssertTest.test_assert_true_branch"
+          docker exec -u user ${container_name} /bin/bash -c "kevm foundry-show --foundry-project-root foundry --verbose AssertTest.test_assert_true_branch"
+          docker exec -u user ${container_name} /bin/bash -c "kevm foundry-list --foundry-project-root foundry --verbose"
+          docker stop --time=0 ${container_name}
```
[the remark this answers is not in the document]
Comment (ehildenb) [reply .github/workflows/test-pr.yml:264]: Issue created; https://github.com/runtimeverification/evm-semantics/issues/1889

File: .github/workflows/test-pr.yml [hunk 3]
```diff
@@ -204,3 +204,61 @@ jobs:
         run: GC_DONT_GC=1 nix build --extra-experimental-features 'nix-command flakes' --print-build-logs
       - name: 'Test KEVM'
         run: GC_DONT_GC=1 nix build --extra-experimental-features 'nix-command flakes' --print-build-logs .#kevm-test
+
+  dockerhub-image-dry-run:
+    name: 'Build Ubuntu Jammy DockerHub Image'
+    runs-on: [self-hosted, linux, normal]
```
Comment (ehildenb) [inline .github/workflows/test-pr.yml:210]:
```suggestion
    name: 'Build Ubuntu Jammy DockerHub Image'
    runs-on: [self-hosted, linux, normal]
    needs: kevm-pyk-code-quality-checks
```

**Pull request merged**